### DISCUSSION TITLE
Cannot save empty values for additional locales

### DISCUSSION THREAD

**Discussion by ivanmart** [thedevdojo/voyager#5403] [Q&A]
Hi, can anybody explain the purpose of lines 232-234 of the Translatable trait?

```
            if (empty($translations[$locale])) {
                continue;
            }
```

It skips empty values, so it becomes impossible to clear any field for additional locales. For example: I have 2 locales: 'en' and 'es'. I can save empty value for `body` or `title` field in Post model for default 'en' locale, but I cannot do that with the same fields when I switch to 'ES' locale at the top right side of Admin panel, it keeps old values after saving. If I remove those lines in Translatable trait - it successfully saves the empty values for the 'es' locale.

**Comment by MrCrayon** [comment]
If you select that line and click on the three dots then click on _View git blame_ you can see when that code was last changed.
In this case it was to solve #4048 with this PR #4049

You can create another Translatable trait that extends Voyager one and change that check.

**Reply by ivanmart** [reply]
@MrCrayon thanks for the explanation! I see, so the main issue in #4048 was that the fallback to default language couldn't be possible due to empty values in translations table. OK, but I have this weird behaviour described by @ghost in #4049: "if you update existing post without providing translation, english values will be applied to the translations table". What can be done about that? I work on a project where some articles (posts) shouldn't be translated and thus shown in the list of posts, so I use `->whereTranslation('body', '!=', '', \App::getLocale())` to select posts from a model. Is there better solution?

I also would be grateful for pointing me to a trait overriding method which can be called from the overridden trait as I can't find a working solution so far.

**Reply by MrCrayon** [reply]
I'm not super familiar with translations in Voyager, if your method is working it looks good enough.

For your question about Trait overriding I am not sure I understand what you mean.
You can create a Trait with the same name in your App that extends Voyager one and then use your Trait in Models.

```php
<?php

namespace App\Traits;

trait Translatable extends \TCG\Voyager\Traits
{
    public function setAttributeTranslations($attribute, array $translations, $save = false)
    {
        // your code
    }
}
```
